sdk/iot/hub/tests/cmocka/test_az_iot_hub_client_methods.c

```diff
@@ -28,17 +28,15 @@ static const az_span test_device_id = AZ_SPAN_LITERAL_FROM_STR(TEST_DEVICE_ID_ST
 static uint8_t g_expected_methods_subscribe_topic[] = "$iothub/methods/POST/#";
 
 #ifndef NO_PRECONDITION_CHECKING
-
 enable_precondition_check_tests()
 
-    static void test_az_iot_hub_client_methods_subscribe_topic_filter_get_NULL_client_fail(
```

Comment:
Can we do anything about this odd indentation that occurs whenever we have `enable_precondition_check_tests()` without a semi-colon? Is there some formatting issue with that `#define`?

https://github.com/Azure/azure-sdk-for-c/blob/35629203f7dab143a65e55760f10ef211f204e1d/sdk/core/core/test/inc/az_test_precondition.h#L25-L32

Putting a semi-colon after it solved the formatting issue, but some  compilers don't like that.

https://dev.azure.com/azure-sdk/public/_build/results?buildId=356698&view=logs&jobId=5e607c67-84c7-5489-c4f9-bbc2a3f1af84&j=5e607c67-84c7-5489-c4f9-bbc2a3f1af84&t=6f51c232-3573-5b2b-2452-94e46b1c30c5
```text
/home/vsts/work/1/s/sdk/iot/hub/tests/cmocka/test_az_iot_hub_client_sas.c:30:34: error: ISO C does not allow extra ‘;’ outside of a function [-Werror=pedantic]
 enable_precondition_check_tests();
                                  ^
cc1: all warnings being treated as errors
```

cc @ewertons, @barcharcraz

Reply:
yea I had noticed this but couldn't find a solution that the compiler liked.